components/landing/topCollections/utils/useTopCollections.ts

```diff
@@ -0,0 +1,91 @@
+import { sanitizeIpfsUrl } from '@/components/rmrk/utils'
+import {
+  calculateAvgPrice,
+  monthlyVolume,
+  monthlyrangeVolume,
+  threeMonthRangeVolume,
+  threeMonthlyVolume,
+  volume,
+  weeklyVolume,
+  weeklyrangeVolume,
+} from '@/components/series/utils'
+import {
+  CollectionEntity,
+  CollectionEntityWithVolumes,
+  CollectionSales,
+} from './types'
+import { useTopCollectionList } from './useTopCollectionList'
+import { useCollectionsSales } from './useCollectionsSales'
+
+const proccessData = (
+  collectionEntities: CollectionEntity[],
+  collectionsSales: CollectionSales[]
+) => {
+  return collectionEntities.map((e): CollectionEntityWithVolumes => {
+    const thisCollectionSales = collectionsSales.find(
+      ({ id }) => id === e.id
+    ) as CollectionSales
+    const saleEvents = thisCollectionSales.sales.map((nft) => nft.events).flat()
+
+    return {
+      ...e,
+      image: sanitizeIpfsUrl(e.image),
+      averagePrice: calculateAvgPrice(e.volume as string, e.buys),
+      volume: volume(saleEvents),
+      weeklyVolume: weeklyVolume(saleEvents),
+      monthlyVolume: monthlyVolume(saleEvents),
+      threeMonthVolume: threeMonthlyVolume(saleEvents),
+      weeklyrangeVolume: weeklyrangeVolume(saleEvents),
+      monthlyrangeVolume: monthlyrangeVolume(saleEvents),
+      threeMonthlyrangeVolume: threeMonthRangeVolume(saleEvents),
+    }
+  })
+}
+
+export const useTopCollections = (limit: number) => {
+  const topCollectionWithVolumeList = useState<CollectionEntityWithVolumes[]>(
+    'topCollectionWithVolumeList',
+    () => []
+  )
+  const fetchError = ref(null)
+  const isLoading = ref(true)
+  const { data: topCollections, error: fetchTopCollectionsError } =
+    useTopCollectionList(limit)
+  watch(
+    [topCollections, fetchTopCollectionsError],
+    ([collectionEntities, fetchTopCollectionsErrorValue]) => {
+      if (fetchTopCollectionsErrorValue) {
+        fetchError.value = fetchTopCollectionsErrorValue
+        isLoading.value = false
+        return
+      }
+      if (collectionEntities) {
+        const { data: collectionSales, error: fetchCollectionsSalesError } =
+          useCollectionsSales(collectionEntities.map((c) => c.id))
+
+        watch(
```

Comment:
nested watch? can't we do otherwise?

Reply:
I thought it was easier to read this way
can move the internal watch outside if you think that's better

Reply:
I think the way to avoid this would be fetch all the necessary data in single query